controller/characterController.go

```diff
@@ -0,0 +1,83 @@
+package controller
+
+import (
+	"os"
+	"encoding/csv"
+	"strconv"
+	"io"
+	
+	"github.com/carRub/academy-go-q12021/model"
+	"github.com/carRub/academy-go-q12021/usecase"
+)
+
+// CharacterController defines what a controller must contain
+type CharacterController struct {
+	characterInteractor usecase.CharacterInteractor
+}
+
+// CharacterUseCase defines functions to refer to our use cases
+type CharacterUseCase interface {
+	GetCharacters() ([]model.Character)
+	GetCharacterByID(id int) (*model.Character) 
+}
+
+// NewCharacterController Creates a new character controller 
+func NewCharacterController (ci usecase.CharacterInteractor) CharacterController {
+	return CharacterController{ci}
+}
+
+func (c CharacterController) GetCharacters() ([]model.Character, error) {
+	csvFile, _ := os.Open("assets/characters.csv")
+	defer csvFile.Close()
+	r := csv.NewReader(csvFile)
+
+	var character model.Character
+	var characters []model.Character
+
+	for {
+		record, err := r.Read()
+		
+		if err == io.EOF {
```

Comment:
As I recommended to use wrapped errors, this checking might fail.

Use `errors.Is()` method to check for a specific error when dealing with wrapped errors.

Reply:
This applies to the GetCharacterByID() method.